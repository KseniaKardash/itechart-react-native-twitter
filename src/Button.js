import React from "react";
import { TouchableHighlight, StyleSheet, Text } from "react-native";

const Button = (props: Props) => {
  const { text, onPress } = props;
  return (
    <TouchableHighlight style={styles.buttonContainer}>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableHighlight>
  );
};
export default Button;

const styles = StyleSheet.create({
  buttonContainer: {
    backgroundColor: "#ff5d39",
    margin: 5,
    padding: 5,
    marginBottom: 10,
    marginTop: 10,
    borderRadius: 10
  },
  buttonText: {
    textAlign: "center",
    color: "#ffffff",
    fontSize: 18,
    padding: 5
  }
});
