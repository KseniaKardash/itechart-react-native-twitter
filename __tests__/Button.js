import React from "react";
import { shallow, mount, configure } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Button from "../src/Button";

configure({ adapter: new Adapter() });

describe("Button snapShot enzyme", () => {
  it("renders BG color as expected", () => {
    const wrapper = shallow(<Button />);
    const color = "#ff5d39";
    expect(wrapper.prop("style").backgroundColor).toEqual(color);
  });

  it("renders props as expected", () => {
    const wrapper = shallow(<Button />);
    wrapper.setProps({ text: "hi" });
    expect(wrapper.prop("children").props.children).toEqual("hi");
  });

  it("renders as expected", () => {
    const wrapper = shallow(<Button text={"t"} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ text: "t" });
    expect(wrapper).toMatchSnapshot();
  });
});
