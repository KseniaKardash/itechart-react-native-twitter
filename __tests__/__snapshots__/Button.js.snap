// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapShot enzyme renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="t"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.85,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 18,
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        t
      </Text>,
      "delayPressOut": 100,
      "style": Object {
        "backgroundColor": "#ff5d39",
        "borderRadius": 10,
        "margin": 5,
        "marginBottom": 10,
        "marginTop": 10,
        "padding": 5,
      },
      "underlayColor": "black",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "t",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#ffffff",
          "fontSize": 18,
          "padding": 5,
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "t",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.85,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          t
        </Text>,
        "delayPressOut": 100,
        "style": Object {
          "backgroundColor": "#ff5d39",
          "borderRadius": 10,
          "margin": 5,
          "marginBottom": 10,
          "marginTop": 10,
          "padding": 5,
        },
        "underlayColor": "black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "t",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#ffffff",
            "fontSize": 18,
            "padding": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "t",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button snapShot enzyme renders as expected 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="t"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.85,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 18,
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        t
      </Text>,
      "delayPressOut": 100,
      "style": Object {
        "backgroundColor": "#ff5d39",
        "borderRadius": 10,
        "margin": 5,
        "marginBottom": 10,
        "marginTop": 10,
        "padding": 5,
      },
      "underlayColor": "black",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "t",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#ffffff",
          "fontSize": 18,
          "padding": 5,
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "t",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.85,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          t
        </Text>,
        "delayPressOut": 100,
        "style": Object {
          "backgroundColor": "#ff5d39",
          "borderRadius": 10,
          "margin": 5,
          "marginBottom": 10,
          "marginTop": 10,
          "padding": 5,
        },
        "underlayColor": "black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "t",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#ffffff",
            "fontSize": 18,
            "padding": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "t",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
